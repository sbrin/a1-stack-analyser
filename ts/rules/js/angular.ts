import { register } from '../../register.js';

register({
  tech: 'angular',
  name: 'Angular',
  type: 'language',
  files: ['angular.json'],
  dependencies: [
    { type: 'npm', name: 'angular' },
    { type: 'npm', name: /^@angular\//, example: '@angular/forms' },
  ],
});
