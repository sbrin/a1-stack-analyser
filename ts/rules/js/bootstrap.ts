import { register } from '../../register.js';

register({
  tech: 'bootstrap',
  name: 'Bootstrap',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'bootstrap' }],
});
