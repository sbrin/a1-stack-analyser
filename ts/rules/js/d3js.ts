import { register } from '../../register.js';

register({
  tech: 'd3js',
  name: 'D3.js',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'd3' },
    { type: 'npm', name: 'd3-shape' },
    { type: 'npm', name: '@types/d3' },
    { type: 'npm', name: '@types/d3-shape' },
  ],
});
