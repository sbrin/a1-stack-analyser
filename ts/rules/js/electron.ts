import { register } from '../../register.js';

register({
  tech: 'electron',
  name: 'Electron',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'electron' },
    { type: 'githubAction', name: 'samuelmeuli/action-electron-builder' },
  ],
});
