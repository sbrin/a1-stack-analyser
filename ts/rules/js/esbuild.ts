import { register } from '../../register.js';

register({
  tech: 'esbuild',
  name: 'Esbuild',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'esbuild' }],
});
