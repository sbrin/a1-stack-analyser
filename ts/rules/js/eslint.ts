import { register } from '../../register.js';

register({
  tech: 'eslint',
  name: 'Eslint',
  type: 'tool',
  files: ['.eslintrc', '.eslintrc.cjs', '.eslintrc.json', '.eslintrc.js'],
  dependencies: [{ type: 'npm', name: 'eslint' }],
});
