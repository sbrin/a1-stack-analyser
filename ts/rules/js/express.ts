import { register } from '../../register.js';

register({
  tech: 'express',
  name: 'Express',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'express' }],
});
