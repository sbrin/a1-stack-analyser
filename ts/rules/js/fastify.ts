import { register } from '../../register.js';

register({
  tech: 'fastify',
  name: 'Fastify',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'fastify' },
    { type: 'npm', name: '@opentelemetry/instrumentation-fastify' },
  ],
});
