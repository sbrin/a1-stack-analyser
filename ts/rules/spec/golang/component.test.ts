import { describe, expect, it } from 'vitest';

import { analyser } from '../../../analyser/index.js';
import { flatten } from '../../../payload/helpers.js';
import { FakeProvider } from '../../../provider/fake.js';
import { cleanNestedSnapshot } from '../../../tests/helpers.js';

import '../../../autoload.js';

describe('golang', () => {
  it('should match a service', async () => {
    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['cmd/', 'go.mod'],
          '/cmd': ['server/'],
          '/cmd/server': ['main.go'],
        },
        files: {
          '/go.mod': `module github.com/specfy/specfy`,
          '/cmd/main.go': `hello world`,
        },
      }),
    });
    expect(cleanNestedSnapshot(flatten(res).toJson())).toMatchSnapshot();
  });
});
