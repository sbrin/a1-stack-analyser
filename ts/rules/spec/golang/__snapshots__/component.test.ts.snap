// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`golang > should match a service 1`] = `
{
  "childs": [
    {
      "childs": [],
      "dependencies": [],
      "edges": [],
      "id": "",
      "inComponent": "",
      "languages": {
        "Go": 1,
      },
      "name": "server",
      "path": [
        "/cmd/server/main.go",
      ],
      "reason": [],
      "tech": "golang",
      "techs": [
        "golang",
      ],
    },
  ],
  "dependencies": [],
  "edges": [],
  "id": "",
  "inComponent": "",
  "languages": {},
  "name": "flatten",
  "path": [
    "/",
  ],
  "reason": [],
  "tech": null,
  "techs": [],
}
`;
